compressed_image_transport/src/compressed_subscriber.cpp

```diff
@@ -17,6 +17,33 @@ namespace enc = sensor_msgs::image_encodings;
 namespace compressed_image_transport
 {
 
+void CompressedSubscriber::subscribeImpl(ros::NodeHandle& nh, const std::string& base_topic, uint32_t queue_size,
+                             const Callback& callback, const ros::VoidPtr& tracked_object,
+                             const image_transport::TransportHints& transport_hints)
+{
+    typedef image_transport::SimpleSubscriberPlugin<sensor_msgs::CompressedImage> Base;
+    Base::subscribeImpl(nh, base_topic, queue_size, callback, tracked_object, transport_hints);
+
+    // Set up reconfigure server for this topic
+    reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
+    ReconfigureServer::CallbackType f = boost::bind(&CompressedSubscriber::configCb, this, _1, _2);
+    reconfigure_server_->setCallback(f);
+}
+
+
+void CompressedSubscriber::configCb(Config& config, uint32_t level)
+{
+  config_ = config;
+  if (config_.mode == "gray") {
+      flags_ = CV_LOAD_IMAGE_GRAYSCALE;
```

Comment:
Thanks, I will do that.

On Mon, Apr 27, 2015 at 2:01 PM, Vincent Rabaud <notifications@github.com>
wrote:

> In compressed_image_transport/src/compressed_subscriber.cpp
> <https://github.com/ros-perception/image_transport_plugins/pull/16#discussion_r29140116>
> :
>
> > +{
> > +    typedef image_transport::SimpleSubscriberPlugin<sensor_msgs::CompressedImage> Base;
> > +    Base::subscribeImpl(nh, base_topic, queue_size, callback, tracked_object, transport_hints);
> > +
> > +    // Set up reconfigure server for this topic
> > +    reconfigure_server_ = boost::make_shared<ReconfigureServer>(this->nh());
> > +    ReconfigureServer::CallbackType f = boost::bind(&CompressedSubscriber::configCb, this, _1, _2);
> > +    reconfigure_server_->setCallback(f);
> > +}
> > +
> > +
> > +void CompressedSubscriber::configCb(Config& config, uint32_t level)
> > +{
> > +  config_ = config;
> > +  if (config_.mode == "gray") {
> > +      flags_ = CV_LOAD_IMAGE_GRAYSCALE;
>
> please use cv::IMREAD_GRAYSCALE` and such from
> https://github.com/Itseez/opencv/blob/2.4/modules/highgui/include/opencv2/highgui/highgui.hpp#L165
> This make it compilable with OpenCV3 too
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ros-perception/image_transport_plugins/pull/16/files#r29140116>
> .
>



-- 
Cedric Pradalier